src/System.Runtime.Caching/src/System/Runtime/Caching/MemoryMonitor.cs

```diff
@@ -42,15 +42,14 @@ internal abstract class MemoryMonitor
         private static long s_totalPhysical;
         private static long s_totalVirtual;
 
-        [SecuritySafeCritical]
         static MemoryMonitor()
         {
-            MEMORYSTATUSEX memoryStatusEx = new MEMORYSTATUSEX();
-            memoryStatusEx.Init();
-            if (UnsafeNativeMethods.GlobalMemoryStatusEx(ref memoryStatusEx) != 0)
+            Interop.Kernel32.MEMORYSTATUSEX memoryStatusEx;
+            memoryStatusEx.dwLength = (uint)Marshal.SizeOf(typeof(Interop.Kernel32.MEMORYSTATUSEX));
+            if (Interop.Kernel32.GlobalMemoryStatusEx(out memoryStatusEx) != 0)
```

Comment:
Should this be ref, rather than out, since you're passing it in?

Reply:
well, yes... Two reasons why it isn't here:
1. Interop.Kernel32.GlobalMemoryStatusEx is already shared by others and unfortunately it is defined with the 'out' parameter like this: GlobalMemoryStatusEx(out MEMORYSTATUSEX).
2. il has both out and ref as ref type so even though from C# syntax style it is better to define it ref, both would technically work.

Besides making it ref would require allocating MEMORYSTATUSEX on the heap (because I could not find an easy way for C# to allow me to use a stack-allocated struct as a ref parameter).